### DISCUSSION TITLE
how to get message mention on reply

### DISCUSSION THREAD

**Discussion by ArthurSStrong** [Rapptz/discord.py#6572] [Q&A] [answered]
i have been searching in documentation a way to get the mention id on reply

i mean the event when a user reply to another message using the "reply button"

i used this code

`user_id= context.message.mentions[0].id`

but it seems to only works when you reply to bot

**Comment by lgaan** [comment, marked as answer]
You will be able to use `discord.Message.reference` in order to get the `MessageReference` object. This represents the data of the reply. From here you can either use `MessageReference.cached_message` if the message is in cache, or fetch the message using `MessageReference.message_id`. This allows you to use `message.author.id` as normal.

**Comment by ArthurSStrong** [comment]
For anyone looking for the code here's it
```
if context.message.reference is None:
        return

 print(context.message.reference.message_id)

 reference_msg = await context.fetch_message(context.message.reference.message_id)
```
i'm closing the discussion thanks @lgaan
